### PR TITLE
groups modifications

### PR DESCRIPTION
**Pull request by rcordeau**
This change is [Reviewable](https://reviewable.io/reviews/smarterapp/rdw_ingest/315)

### CODE REVIEW

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java [hunk 1]
```diff
@@ -50,25 +60,35 @@ public void loadBatch(final String digest, final long batchId) {
...
-            if(uri.startsWith(s3Prefix)) {
-                sql = sqlListBuilder.addNext("load", "fromS3").build();
-            }
-            else {
-                sql = sqlListBuilder.addNext("load", "fromLocal").build();
-            }
-            repository.execute(sql, ImmutableMap.of("uri", uri,
-                    "batch_id", batchId));
+        if (!archiveService.exists(location)) {
+            logger.debug("Unable to locate " + location + " via the archive service");
```
Comment (jtreuting) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: if this is throwing an exception, shouldn't that be a logger.error?

Comment (rcordeau) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: Yes, good call.

Comment (wtritch) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: per @malaffoon logger.warn might be more appropriate?

Comment (malaffoon) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: log-and-throw is an antipattern; will the exception be logged further up?

Comment (rcordeau) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: I made it as .error for now.

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java [hunk 2]
```diff
@@ -50,25 +60,35 @@ public void loadBatch(final String digest, final long batchId) {
...
+        }
+        else if(uri.startsWith(localPrefix)){
+            sqlSource = "LOCAL INFILE";
+            uri = uri.substring(localPrefix.length());
+        }
+        else {
+            throw new IllegalArgumentException("Invalid URI for location " + location);
+        }
+
+        List<String> sql = sqlListBuilder.addNext(String.format(rawSql, sqlSource)).build();
```
Comment (malaffoon) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: This'll work but i have two concerns: String.format can do unexpected things, and %s placeholder is pretty non-specific. I'd lean towards putting `%%sqlSource%%` in the sql string resource and doing a substr() here.

Comment (rcordeau) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: Will fix.

Comment (rcordeau) [reply group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultProcessingLoadService.java]: How about replace(s1, s2)?

**Pull request merged**